great_expectations/zep/postgres_datasource.py

```diff
@@ -49,32 +50,11 @@ def param_names(self) -> List[str]:
 
 
 class TableAsset(DataAsset):
-    # Instance variables
+    # Instance fields
+    type: Literal["table"] = "table"
     table_name: str
-    column_splitter: Optional[ColumnSplitter]
-    _name: str
-    _datasource: Datasource
-
-    def __init__(
-        self,
-        name: str,
-        datasource: Datasource,
-        table_name: str,
-        column_splitter: Optional[ColumnSplitter] = None,
-    ) -> None:
-        super().__init__(name)
-        self.table_name = table_name
-        self._datasource = datasource
-        self._name = name
-        self.column_splitter: Optional[ColumnSplitter] = column_splitter
-
-    @property
-    def name(self) -> str:
-        return self._name
-
-    @property
-    def datasource(self) -> Datasource:
-        return self._datasource
+    column_splitter: Optional[ColumnSplitter] = None
+    name: str
```

Comment:
Yes, it can be modified.

We could protect it with pydantic `field` settings.
https://pydantic-docs.helpmanual.io/usage/schema/#field-customization

```python
from pydantic import Field


class TableAsset(DataAsset):
    # Instance fields
    type: Literal["table"] = "table"
    table_name: str
    column_splitter: Optional[ColumnSplitter] = None
    name: str = Field(..., allow_mutation=False)
```